### PR TITLE
enhance: Invalidate shard leader cache when leader location changed

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: #28898
This PR introduce the policy to invalidate shard leader cache by query coord when leader location changed.

### CODE REVIEW

File: internal/querycoordv2/dist/dist_handler.go [hunk 1]
```diff
@@ -209,7 +214,41 @@ func (dh *distHandler) updateLeaderView(resp *querypb.GetDataDistributionRespons
 		updates = append(updates, view)
 	}
 
+	oldViews := dh.dist.LeaderViewManager.GetLeaderView(resp.GetNodeID())
 	dh.dist.LeaderViewManager.Update(resp.GetNodeID(), updates...)
+	dh.TryInvalidateShardLeaderCache(resp.GetNodeID(), oldViews)
+}
```
Comment (congqixia) [inline internal/querycoordv2/dist/dist_handler.go:220]: this is triggered at node level
maybe collection level is better?

File: pkg/util/paramtable/component_param.go [hunk 2]
```diff
@@ -1649,6 +1651,14 @@ func (p *queryCoordConfig) init(base *BaseTable) {
...
 	p.CheckNodeSessionInterval.Init(base.mgr)
+
+	p.ShardLeaderCacheInterval = ParamItem{
+		Key:          "proxy.shardLeaderCacheInterval",
+		Version:      "2.3.4",
+		DefaultValue: "3",
+		Doc:          "time interval to update shard leader cache, in seconds",
+	}
+	p.ShardLeaderCacheInterval.Init(base.mgr)
 }
```
Comment (congqixia) [inline pkg/util/paramtable/component_param.go:1662]: this new config is never used?

**Comment by weiliu1031** [comment]
rerun ut

**Comment by weiliu1031** [comment]
rerun ut

**Pull request closed without merging**